.github/workflows/lint.yaml

```diff
@@ -1,16 +1,26 @@
 name: Lint
 
-on: push 
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+    - planning-1.0-release
+  workflow_dispatch:
 
 jobs:
-  ruff:
+  precommit:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
           python-version: '3.10'
-      - name: Run Ruff
-        uses: chartboost/ruff-action@v1
-        with: 
-          version: 0.5.2
+          
+      - name: Run pre-commit hooks
+        run: |
+          pip install pre-commit
```

Comment:
Why do we need pip here? Is it possible to use poetry and dependency from the `pyproject`?

Reply:
Yes, that is definitely possible. However, AFAIK it is not necessary; `pre-commit` hooks do not need the virtual environment, they run in their own virtual environment. The alternative to the above would be to install `poetry` and the development dependencies, but that takes about 40-45 seconds, whereas the proposal here takes about 8-10 seconds.

Reply:
Clear, thank you for the explanation!